routes/books.js

```diff
@@ -48,56 +91,29 @@ router.post('/', (req, res, next) => {
 })
 
 router.put('/:id', (req, res, next) => {
-  // Make sure that id is a number
-  if (!req.params.id || Number.isNaN(req.params.id)) {
-    throw new Error('Invalid ID')
-  }
-
-  knex('books')
-    .where('id', req.params.id)
-    .first()
-    .then(book => {
-      if (!book) {
-        return next()
-      }
-
-      return knex('books')
-        .update({
-          title: req.body.title,
-          author: req.body.author,
-          genre: req.body.genre,
-          description: req.body.description,
-          cover_url: req.body.coverUrl,
-        }, '*')
-        .where('id', req.params.id)
-    })
-    .then(books => {
-      if (!books) { return {message: 'nothing found for given id'} }
-      const newObj = {
-        id: books[0].id,
-        title: books[0].title,
-        author: books[0].author,
-        genre: books[0].genre,
-        description: books[0].description,
-        coverUrl: books[0].cover_url,
-      }
-      res.send(newObj)
-    })
-    .catch(err => {
-      next(err)
-    })
+  return Promise
+  .resolve(checkInput(req.params))
+  .then(getOne)
+  .then(book => {
+    if (!book || book.length === 0) {
+      return res.status(404).send({message: 'Not found'})
+    }
+    return req.body
+  })
+  .then(update)
+  .then(books => {
+    formatBook(books)
+    res.send(newObj)
+  })
+  .catch(err => {
```

Comment:
Now that `formatBook()` is extracted, we can plug it into the Promise Chain, like so:

```js
//.....
    }
    return req.body
  })
  .then(update)
  .then(formatBook)        //  <----  formatBook can be fed into the `.then` chain directly!
  .then(book => res.send(book))
  .catch(next)
```